File: programming files/sum_even_fibonacci.cc
#include <iostream>

int main()
{
    // Fibonacci base cases:
    int a = 0, b = 1;
    int sum_even = 0;
    // Calculate fib numbers 0 to 4000000
    while (b < 4000000) {
        if (b % 2 == 0) {
            sum_even += b;
        }
        int temp = b;
        b = a + b;
        a = temp;
    }
    std::cout << sum_even << std::endl;

    return 0;
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.java
public class sum_even_fibonacci {
    public static void main(String[] args) {
        int a = 0, b = 1; //a is first in fibanachi sequence b is second in fibonachi sequence
        int sumEven = 0;
        while (b < 4000000) { 
            if (b % 2 == 0) { //checks if b is even and if so it adds it to even sum 
                sumEven += b;
            }
            int temp = b; //temporary variable that holds b 
            b = a + b;
            a = temp; // sets a as temporary variable
        }
        System.out.println(sumEven); //prints the sum 
    }
}

// answer should be 4613732
